enhancements/per-node-configuration.md

```diff
@@ -0,0 +1,484 @@
+kind: Underlay
+metadata:
+  name: underlay-rack-1
+  namespace: openperouter-system
+spec:
+  nodeSelector:
+    matchLabels:
+      topology.kubernetes.io/rack: rack-1
+  asn: 64512
+  evpn:
+    vtepcidr: 100.65.1.0/24
+  nics:
+    - toswitch
+  neighbors:
+    - asn: 64500
+      address: 192.168.1.254  # ToR switch for rack 1
+      bfd:
+        receiveInterval: 300
+        transmitInterval: 300
+---
+# Rack 2 nodes connect to ToR 2
+apiVersion: openpe.openperouter.github.io/v1alpha1
+kind: Underlay
+metadata:
+  name: underlay-rack-2
+  namespace: openperouter-system
+spec:
+  nodeSelector:
+    matchLabels:
+      topology.kubernetes.io/rack: rack-2
+  asn: 64512
+  evpn:
+    vtepcidr: 100.65.2.0/24
+  nics:
+    - toswitch
+  neighbors:
+    - asn: 64500
+      address: 192.168.2.254  # ToR switch for rack 2
+      bfd:
+        receiveInterval: 300
+        transmitInterval: 300
+```
+
+#### Multi-Datacenter Configuration
+
+Different datacenters with different ASNs:
+
+```yaml
+# Underlay for datacenter-east nodes
+apiVersion: openpe.openperouter.github.io/v1alpha1
+kind: Underlay
+metadata:
+  name: underlay-dc-east
+  namespace: openperouter-system
+spec:
+  nodeSelector:
+    matchLabels:
+      topology.kubernetes.io/zone: us-east-1a
+  asn: 64512
+  evpn:
+    vtepcidr: 100.65.0.0/24
+  nics:
+    - eth1
+  neighbors:
+    - asn: 64500
+      address: 192.168.10.1  # ToR switch in DC East
+---
+# Underlay for datacenter-west nodes
+apiVersion: openpe.openperouter.github.io/v1alpha1
+kind: Underlay
+metadata:
+  name: underlay-dc-west
+  namespace: openperouter-system
+spec:
+  nodeSelector:
+    matchLabels:
+      topology.kubernetes.io/zone: us-west-1a
+  asn: 64513
+  evpn:
+    vtepcidr: 100.66.0.0/24
+  nics:
+    - eth1
+  neighbors:
+    - asn: 64501
+      address: 192.168.20.1  # ToR switch in DC West
+```
+
+#### Hardware-Specific NIC Configuration
+
+Different NIC naming across vendor hardware:
+
+```yaml
+# For Dell servers with specific NIC naming
+apiVersion: openpe.openperouter.github.io/v1alpha1
+kind: Underlay
+metadata:
+  name: underlay-dell-hardware
+  namespace: openperouter-system
+spec:
+  nodeSelector:
+    matchLabels:
+      hardware.vendor: dell
+  asn: 64512
+  evpn:
+    vtepcidr: 100.65.0.0/24
+  nics:
+    - eno1
+    - eno2
+  neighbors:
+    - asn: 64500
+      address: 192.168.10.1
+---
+# For HP servers with different NIC naming
+apiVersion: openpe.openperouter.github.io/v1alpha1
+kind: Underlay
+metadata:
+  name: underlay-hp-hardware
+  namespace: openperouter-system
+spec:
+  nodeSelector:
+    matchLabels:
+      hardware.vendor: hp
+  asn: 64512
+  evpn:
+    vtepcidr: 100.65.0.0/24
+  nics:
+    - em1
+    - em2
+  neighbors:
+    - asn: 64500
+      address: 192.168.10.1
+```
+
+### Controller Implementation
+
+#### Node Matching Logic
+
+The Underlay controller will be enhanced with:
+
+1. **Node Label Watching**: Add a watch for Node resource label changes to trigger reconciliation
```

Comment:
we can pinpoint the watcher to these specific bits, right ?

Reply:
> we can pinpoint the watcher to these specific bits, right ?

What do you mean ?